run-clang-format.py

```diff
@@ -0,0 +1,408 @@
+#!/usr/bin/env python
```

Comment:
**suggestion:** Can we move this file under the `scripts` folder and make it executable? It helps to declutter the root directory.

```shell
chmod +x run-clang-format.py
```